components/auth/form/TermsForm.tsx

```
@@ -0,0 +1,97 @@
+
+import BackButton from "@/components/commons/button/BackButton";
+import PrimaryButton from "@/components/commons/button/PrimaryButton";
+import ConsentItem from "@/components/commons/consent/ConsentItem";
+import { AGREEMENTS } from "@/constants/terms";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+
+const TermsForm = () => {
+  const router = useRouter();
+  const [checkedMap, setCheckedMap] = useState<Record<string, boolean>>(
+    AGREEMENTS.reduce((acc, { id }) => ({ ...acc, [id]: false }), {})
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)